src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -0,0 +1,571 @@
+    logger().debug(
+       "{}: *this {} entry {} into left {} right {}", __func__,
+       *this, *entry, *left, *right);
+    auto left_key = iter.get_node_key();
+    left_key.laddr = left->get_laddr();
+    journal_inner_update(iter, left_key, maybe_get_delta_buffer());
+    return checking_parent(oc, pivot, iter+1, this)
+      .safe_then([this, oc, left = left, right = right, pivot = pivot, entry] (auto pair) {
+      auto [extent, iter] = pair;
+      omap_inner_key_t right_key;
+      right_key.laddr = right->get_laddr();
+      right_key.key_len = pivot.size() + 1;
+      right_key.key_off = iter.get_index() == 0 ?
+                          right_key.key_len :
+                          (iter - 1).get_node_key().key_off + pivot.size() + 1;
+      extent->journal_inner_insert(iter, right_key, pivot,
+                       extent->maybe_get_delta_buffer());
+      //retire extent
+      return oc.tm.dec_ref(oc.t, entry->get_laddr())
+        .safe_then([left, right, pivot] (auto ret) {
+        return make_split_entry_ret(
+          make_split_entry_ertr::ready_future_marker{},
+          std::make_tuple(left, right, pivot));
+      });
+    });
+  });
+
+}
+
+OMapInnerNode::split_entry_ret
+OMapInnerNode::split_entry(omap_context_t oc, std::string &key,
+                      internal_iterator_t iter, OMapNodeRef entry)
+{
+  logger().debug("{}: {} key = {}","OMapInnerNode",  __func__, key);
+  return make_split_entry(oc, key, iter, entry).safe_then([&key] (auto tuple) {
+    auto [left, right, pivot] = tuple;
+    return split_entry_ertr::make_ready_future<OMapNodeRef>(
+       pivot > key ? left: right);
+  });
+}
+
+
+OMapInnerNode::merge_entry_ret
+OMapInnerNode::merge_entry(omap_context_t oc, const std::string &key,
+                           internal_iterator_t iter, OMapNodeRef entry)
+{
+  logger().debug("{}: {}","OMapInnerNode",  __func__);
+  if (!is_pending()) {
+    auto mut = oc.tm.get_mutable_extent(oc.t, this)->cast<OMapInnerNode>();
+    auto mut_iter = mut->iter_idx(iter->get_index());
+    mut->parent = this->parent;
+    return mut->merge_entry(oc, key, mut_iter, entry);
+  }
+  logger().debug("{}: {}, {}", __func__, *this, *entry);
+  auto is_left = (iter + 1) == iter_end();
+  auto donor_iter = is_left ? iter - 1 : iter + 1;
+  return omap_load_extent(oc, this, donor_iter->get_node_key().laddr,  get_meta().depth - 1)
+    .safe_then([this, oc, &key, iter, entry, donor_iter, is_left]
+      (auto &&donor) mutable {
+    auto [l, r] = is_left ?
+      std::make_pair(donor, entry) : std::make_pair(entry, donor);
+    auto [liter, riter] = is_left ?
+      std::make_pair(donor_iter, iter) : std::make_pair(iter, donor_iter);
+    if (donor->extent_under_median()) {
+      logger().debug("{}::merge_entry make_full_merge l {} r {}", __func__, *l, *r);
+      return l->make_full_merge(oc, r, this)
+        .safe_then([this, oc, &key, iter, entry, donor_iter, is_left, l, r, liter, riter]
```

Comment:
Please look over this PR and replace long capture lists like this with defaults: [=, &key] in this case.